net/netcheck/netcheck.go

```diff
@@ -875,6 +895,26 @@ func (c *Client) GetReport(ctx context.Context, dm *tailcfg.DERPMap) (_ *Report,
 
 	plan := makeProbePlan(dm, ifState, last)
 
+	// If we're doing a full probe, also check for a captive portal.
```

Comment:
I had assumed we'd only do this when UDP failed. You think STUN could work but port 80 would be intercepted?

Reply:
Not on a "good" captive portal, no, but I have no idea how less-good ones behave, and HTTP requests are pretty cheap.

Reply:
They're cheap but it's a lot of client nodes in the field. Can we at least delay the captive portal check like 50ms and abort it if we get any UDP STUN reply in those 50ms?

Reply:
Added a 200ms delay, chosen semi-randomly as a value that resulted in no captive portal checks firing on my home network.